ansible/roles/polkadot-validator/templates/proxy.conf.j2

```diff
@@ -0,0 +1,4 @@
+server {
+  listen 0.0.0.0:{{ proxy_port }};
+  proxy_pass localhost:{{ p2p_port }};
```

Comment:
I guess in the future we could limit the amount of allowed incoming connections here. In case of a DOS attack trying to exhaust the connection limit only the nginx port would be hit, the Polkadot process itself could still make outgoing connections.

Have you experimented with these limits in the past @fgimenez?

Reply:
Indeed, i've used some of these features before in the context of nginx ingress controllers, mainly limiting the number of connections and requests per second, but at the application layer, this is the full list of available capabilities  https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#rate-limiting There is an equivalent capability for limiting the number of connections per IP in the streams module https://nginx.org/en/docs/stream/ngx_stream_limit_conn_module.html

Reply:
> There is an equivalent capability for limiting the number of connections per IP in the streams module https://nginx.org/en/docs/stream/ngx_stream_limit_conn_module.html

This looks like a good first step to explore. One could configure it on a single instance and then run a TCP load test against it.